src/components/App/AnimatedWrapper.js

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react';
+import { node } from 'prop-types';
+import styled, { withTheme } from 'styled-components';
+import { animated, useTransition, config } from 'react-spring';
+
+import { location, theme } from '../../types';
+import extractRoleFromPath from '../../utils/extractRoleFromPath';
+import generateColors from '../../utils/generateColors';
+
+const Wrapper = styled(animated.div)`
+  position: absolute;
+  width: 100%;
+  min-height: 100%;
+  display: grid;
+  grid-template-rows: auto 1fr auto;
+`;
+
+function usePreviousRole(currentRole) {
```

Comment:
I am not sure about the hooks naming, since that `use` keyword seems strange in most cases for me :D